### PR TITLE
simplify logger API

### PR DESCRIPTION
**Pull request by achille-roussel**
@segmentio/infra 

I've done a bunch of cleaning on the logger API, mostly got rid of a a bunch of high-level stuff. I think the `ecslogs` package will be more useful if it focused on exposing building blocks to work with `ecs-logs` instead of attempting to reinvent the wheel.
There are great logging packages in Go (logrus, apex/log, ...), let's focus on leveraging them by providing an API that makes it easy to build plugins for those packages instead of offering a lower quality version of what they already solved.

Basically here's the logger API now:
```go
type Logger interface {
    Log(Event) error
}
```
Single-method interfaces are very powerful in Go and allow for a bunch of interesting patterns which I'm sure will be more useful.

Please take a look and let me know what you think about this.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -4,12 +4,6 @@
 [submodule "vendor/github.com/coreos/go-systemd"]
 	path = vendor/github.com/coreos/go-systemd
 	url = https://github.com/coreos/go-systemd
-[submodule "vendor/github.com/Sirupsen/logrus"]
-	path = vendor/github.com/Sirupsen/logrus
-	url = https://github.com/Sirupsen/logrus
-[submodule "vendor/github.com/apex/log"]
-	path = vendor/github.com/apex/log
-	url = https://github.com/apex/log
```
Comment (achille-roussel) [inline .gitmodules]: This was necessary because the packages that depend on `ecslogs/logrus` or `ecslogs/apex` would fail to compile otherwise.

File: lib/runtime.go [hunk 2]
```diff
@@ -37,22 +37,25 @@ search:
...
 		}
 
 		for _, pkg := range ignorePackages {
-			if strings.HasPrefix(info.File, pkg) {
+			if strings.Contains(info.File, pkg) {
 				continue search
 			}
 		}
 
 		// Now that we got out of the packages that we wanted to ignore we need
```
Comment (yields) [inline lib/runtime.go]: <3

**Comment by yields** [comment]
lgtm! :)
